### DISCUSSION TITLE
enforcing positive solution using sigmoid fails.

### DISCUSSION THREAD

**Discussion by yoanncheny** [lululxvi/deepxde#1401] [Q&A]
Dear DeepXDE users,

I'm solving the BCRE equations with deepxde. The solutions of this problem are under the form h(x,t) and R(x,t) which are positive or zero. 
To help the training and avoiding unrelevant solutions I aply a sigmoid function for the output layer  : 

`net = dde.nn.FNN([2] + 4 * [50] + [2], ["tanh","tanh","tanh","tanh","sigmoid"], "Glorot normal")

model = dde.Model(data, net)

model.compile("adam", lr=1e-3)
losshistory, train_state =model.train(iterations=500)`

However after training the solution has negative values up to -0.3. 

Has anyone already faced such an issue ?
Regards,
Yoann

**Comment by haison19952013** [comment]
Have you tried `apply_output_transform()` ?, I suggest using Softplus instead of sigmoid.

**Reply by yoanncheny** [reply]
Thx for your answer ! Indeed, by using `apply_output_transform()` the solution remains positive. 
However I'm a bit confused : why did I get negative values by applying the sigmoid function to the output layer ? Is this some sort of bug ?

**Comment by yoanncheny** [comment]
The solution proposed by @haison19952013 allows me to get a positive solution. However the solution is totally wrong. Previously it was rather ok except for the unphysical negative values. Has anyone got some hints for understanding this behaviour ?
